source/common/network/connection_impl.cc

```diff
@@ -743,8 +745,9 @@ void ClientConnectionImpl::connect() {
 
   // The local address can only be retrieved for IP connections. Other
   // types, such as UDS, don't have a notion of a local address.
+  // TODO(fcoras) move to SocketImpl?
```

Comment:
+1 same comment about removing the fd round trip somehow.

Reply:
Thanks for confirming! I guess same should be done for INADDR_ANY binds and 0 ports